src/test/java/core/basesyntax/service/impl/DataProcessorServiceImplTest.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.FruitShopException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DataProcessorServiceImplTest {
+    private static List<String[]> list;
+    private DataProcessorServiceImpl dataProcessorService = new DataProcessorServiceImpl();
+
+    @BeforeClass
+    public static void beforeClass() {
+        list = new ArrayList<>();
+    }
+
+    @Before
+    public void setUp() {
+        list.add(new String[]{"b", "apple", "20"});
+        list.add((new String[] {"s", "apple", "30"}));
+    }
+
+    @After
+    public void tearDown() {
+        list.clear();
+    }
+
+    @Test
+    public void processData_addValidData_Ok() {
+        dataProcessorService.processData(list);
+        Integer actual = Storage.mapFruits.get("apple");
+        Integer expected = 50;
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = FruitShopException.class)

```

Comment:
I think that better to avoid "expected" before annotation better to use @Rule annotation with ExpectedException class 
but mentor know this better it doesn`t need to change

Reply:
ok, i will read about @ rule